components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -155,62 +153,48 @@ public EmailSenderDTO addEmailSender(EmailSenderDTO emailSender) throws Notifica
     @Override
     public SMSSenderDTO addSMSSender(SMSSenderDTO smsSender) throws NotificationSenderManagementException {
 
-        // Set the default publisher name if name is not defined.
-        if (StringUtils.isEmpty(smsSender.getName())) {
-            smsSender.setName(DEFAULT_SMS_PUBLISHER);
-        }
-
-        validateSMSSender(smsSender);
-
-        Optional<Resource> resourceOptional = getPublisherResource(smsSender.getName());
-
-        if (resourceOptional.isPresent()) {
-            throw new NotificationSenderManagementClientException(ERROR_CODE_CONFLICT_PUBLISHER, smsSender.getSender());
+        Map<String, String> properties = smsSender.getProperties();
```

Comment:
can `smsSender.getProperties()` be null?

Reply:
private Map<String, String> properties = new HashMap<>(); 
This is how it is defined in the DTO class. Therefore it cannot be null unless null is assigned. Is it required to ada a null check in that case?